src/launch/launcher.rs

```diff
@@ -61,16 +65,57 @@ impl<'a, U: AsRawFd, V: AsRawFd> Launcher<'a, New, U, V> {
 
         Ok(next)
     }
+
+    /// Begin the SEV-SNP launch process by creating a Launcher and issuing the
+    /// KVM_SNP_INIT ioctl.
+    pub fn snp_new(kvm: &'a mut U, sev: &'a mut V) -> Result<Self> {
```

Comment:
So, I made a real running SNP VM here:
https://github.com/haraldh/sev/blob/real_code/tests/snp_launch.rs

For this to work, I needed a `&mut vm_fd` (here named `kvm`).
[This](https://github.com/haraldh/sev/commit/7d2775847a49f3af5c886fc7c30606bc48427a20) patch enabled it.

Patch for the real snp_launch test:
https://github.com/haraldh/sev/commit/eacfd06360b362c9064847cf28b3f0471c3f8f81

Reply:
Btw, I would love to see a rename from `kvm` to `kvm_vm` or `vm_fd` throughout the crate to distinguish it from the `kvm` fd, just to prevent confusion.